### PR TITLE
Task 2: Unit test for specific functions part 3

### PR DESCRIPTION
**Pull request by Rahul-JOON**
This PR contains the test case for the functions - _calculate_contact_forces_self_rod & _calculate_contact_forces_rod_rod as mentioned in #99.

Because the codebase is not broken, all of the test cases are currently passing.

### CODE REVIEW

**Comment by bhosale2** [comment]
@Rahul-JOON a small note, if you are still making changes to a PR after opening it, please either mark it as WIP or draft unless final to review.

**Comment by Rahul-JOON** [comment]
Understood and noted, Thanks @bhosale2

File: tests/test_contact_specific_functions.py [hunk 1]
```diff
@@ -224,3 +226,180 @@ def test_claculate_contact_forces_rod_rigid_body():
...
+                [0.719324, 1.120859, 1.575353],
+                [1.790064, 0.204983, 1.113846],
+            ]
+        ),
+        rtol=tol,
+        atol=tol,
+    )
+
+
+def test_claculate_contact_forces_self_rod():
```
Comment (armantekinalp) [inline tests/test_contact_specific_functions.py:348]:
```suggestion
def test_calculate_contact_forces_self_rod():
```

**Review by armantekinalp — APPROVED** [review]
LGTM, one minor change.

**Review by bhosale2 — APPROVED** [review]
LGTM

**Pull request merged**